### ISSUE TITLE
(re)loading and setting bibliography takes too long 

### ISSUE THREAD

**Issue by novusshu** [jkitchin/org-ref#967]
Now I updated to Version 3. Every time I open an org file with bibliography, the "_(re)loading bibliography_"  takes at least 10-15 seconds, while in the past it was only 2-3 seconds. I tried with the code provided in this issue https://github.com/jkitchin/org-ref/issues/928, but it didn't help much. 

More, the _bibliography:_ and _biliographystyle:_ lines (for setting the bib file and style) are almost impossible to edit. It takes another 10-15 seconds if I want to kill or add a single letter and this is excruciating! Has anyone experienced the same issue and is there a fix?

Thank you very much!!!

**Comment by jkitchin** [comment]
Can you run `M-x org-ref` in that org-file and put the output here? 

I don't see any thing that extreme on those lines. There is some room to improve it though. I pushed a small improvement on bibliographystyles that should only be slow on the first edit, but after that should be faster.

**Comment by novusshu** [comment]
> 
Below is the output as you instructed:
`funcall-interactively: Wrong type argument: listp, #<marker at 20520 in main.org`

The strange thing is my problem seemed gone as of now (no lag whatsoever, and I don't even see a "(re)loading bibliography" message even in a new file), but I haven't updated any package yet. The only thing changed is the updated .bib file (I use Zotero to auto-update it).

**Comment by jkitchin** [comment]
hm. that is pretty odd, I don't see an error like that on my end. Do you get that error in every org file or just this one?

**Comment by novusshu** [comment]
Yeah it's quite strange. The org-ref report can be generated normally in an org file that does not have the bibliography or bibliographystyle lines. Any file with these lines can only result in the same error and an empty report with a "bad files" entry.
